File: src/lec07/Lec07Main.java
package lec07;

import java.io.IOException;
import org.jetbrains.annotations.NotNull;

public class Lec07Main {

  public static void main(String[] args) throws IOException {
    JavaFilePrinter printer = new JavaFilePrinter();
    printer.readFile();
  }

  private int parseIntOrThrow(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      throw new IllegalArgumentException(String.format("%s는 숫자가 아님", str));
    }
  }

  private Integer parseIntOrNull(@NotNull String str) {
    try {
      return Integer.parseInt(str);
    } catch (NumberFormatException e) {
      return null;
    }
  }
}

File: src/lec07/Lec07Main.kt
package lec07

import java.lang.IllegalArgumentException

fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt() //기본 타입 간의 형변환은 toType()을 사용함
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("${str}은 숫자가 아님")
    }
}

fun parseIntOrNull(str: String): Int? {
    return try {
        str.toInt() //기본 타입 간의 형변환은 toType()을 사용함
    } catch (e: NumberFormatException) {
        null
    }
}